### DISCUSSION TITLE
Scene Matching (Online Episodes) with GEVI?

### DISCUSSION THREAD

**Discussion by ms52538** [CodyBerenson/PGMA-Modernized#155] [General]
First: a big Thank You for all the effort you have put into creating and maintaining these agents.  

I do have an observation/question:

GEVI is not listed specifically under the SCENE matching agent (vs it is available in the agents for Adult/AdultFilms). 

In an effort to match SCENES with the existing resources provided with the Adult Scenes agent, it seems the most relied upon service is IAFD, followed by the 4 blogs sites. More often than not I'm finding that GEVI is a more reliable source of records (equal to or more than IAFD) yet GEVI is not specifically available in the SCENES agent. Am I correct in concluding that GEVI is not being utilized by any of the agents for SCENES (Online Episodes) matching? If so, is there a technical as to why? A considered work-around of force matching using the Adult/Adult Films agents, with GEVI enabled, seems incapable of correctly matching.

For example:   (Men At Play) - Paint (2011) [Ben Brown, Neil Stevens]
The SCENEs agent cannot match the file (neither the agent's blog sites or IAFD database have an entry for the scene) - whereas GEVI does, and includes both a summary AND a cover-art image for the file.

I am often running into this wall and cannot seem to enable a work-around to matching with GEVI Online Episodes records.

Apologies if this has already been noted in the Installation section or raised in the discussion area. 

Thanks!

**Comment by JPH71** [comment]
I've just seen that you mentioned that GEVI now provides poster art...

I will start looking at that...

Cheers xxx

**Comment by CodyBerenson** [comment]
@ms52538 
Hello.  I too frequently use GEVI when creating metadata for last resort matching with the GAnfoMoviesImporter.bundle, so I understand and feel your pain.  

Since you've been navigating and searching with GEVI, you've likely noticed that GEVI doesn't support searching directly for scenes as it does feature titles...one has to navigate to a scene by starting with either a search for the performer or the studio, and drilling down/navigating from there, which isn't how our agents process a search for a match.  Simply put, unless I'm mistaken, there is no search string that gets a results list for GEVI scenes.  If you believe this is not true, please let us know.

Cheers

**Reply by ms52538** [reply]
Thanks for the quick reply. I just noticed that little 'hiccup' earlier today, so I understand exactly the point of limitation you reference. I am curious if you have direct contact with the site owner of GEVI and if there is any efforts in coordinating handshaking between their site code and your agent code?

**Comment by JPH71** [comment]
One would have t code a new routine to search for scenes... the agent
specifically caters only for Full Films rather than scenes.
But as you mentioned if GEVI is now doing poster art, it wouldn't be too
difficult to do a GEVI scene agent...

**Reply by ms52538** [reply]
I would not go so far as to say GEVI is providing what we would call Front and Back cover art images. The example of the Men at Play Paint scene I initially reference, has a graphic image but not the type that is full-fledge cover art (as we associate with the blog sites, and really is the only reason we are forced to use them in matching) as compared to IAFD which has limited or no images to use - so at present I'd say it's a half-way point for GEVI. Which leads me to wonder what scope GEVI hopes to achieve in the future. As it stands presently, GEVI is producing daily 'round-up' of scenes being published, and a more complete concentration each day vs the blog sites (which appear random between who publishes what content on any given day by any of the adult content sites; my chief issue with the blogs is that sooner or later they 'archive' their content, shifting urls, and records disappear whereas GEVI is tried and true). I'm not sure where GEVI's owner wants to land in the mix of things; clearly he has a passion for what he does - the site reflects it, but how far does he want to make GEVI's data accessible in partnering with meta-data agents/developers.

**Comment by JPH71** [comment]
I have just seen what you mean...- one can't search for scenes... darned it

**Comment by JPH71** [comment]
no contact with any of the sites - most would not approve of us scraping
their sites for content, but i will look into it to see if there is a way
to do it....

**Comment by JPH71** [comment]
Gevi put in blocks this year to prevent scraping... they share the same
principle as the other websites.. so we date not involve them or question
them in relation to this...

**Comment by carylee** [comment]
Have you considered adding a manual match option where users could supply the GEVI episode id? For example, maybe if the naming convention were extended to allow for something like "GEVI-episode-180529", then the code could bypass all the searching and jump straight to the episode url and continue with fetching the metadata from there.

Obviously this wouldn't provide the convenience that automatic matching does, but it would provide a way to download metadata for videos that exist in GEVI that people are willing to tag manually.

**Reply by JPH71** [reply]
I don't think Plex allows that - but it can be engineered by putting the ID in the filename.... if it is not too long....

**Comment by JPH71** [comment]
Unfortunately GEVi does not allow you to search for Scenes.... grrr

**Comment by CodyBerenson** [comment]
@ms52538 @carylee 

Great news!  Jason @JPH71 has created a GEVI Scenes matching agent.  I'm testing it now, so stay tuned.  The unique scene # is part of the file name.  The GEVI Scenes agent will scrape the GEVI Scene, return the GEVI Scene Image and rotate it 90 degrees so that it is portrait vs. landscape/cropped. 

![image](https://github.com/CodyBerenson/PGMA-Modernized/assets/40373801/54ffb2cc-d6eb-40c1-a5d7-4b76fd5b8f84)

**Reply by dimmitri9** [reply]
WOAAAAAAAAAAAAAAAAHHHHHHHHHHHHHHHHHHHHHH !!!!!!!!!!!!!!

**Comment by ms52538** [comment]
That is Great news indeed! Crossing fingers for successful testing and
final deployment! Thanks for the follow-up. :) I'll keep an eye out for the
release. Much, MUCH appreciated!

**Comment by dimmitri9** [comment]
oh myyyyyyyy! =DDDDDDDDD
